### ISSUE TITLE
[okta-react] Lifecycle event componentWillMount on RenderWrapper generates warning message for React 16.9.0

### ISSUE THREAD

**Issue by dbryantm** [okta/okta-oidc-js#532]
## I'm submitting this issue for the package(s):

- [ ] jwt-verifier
- [ ] okta-angular
- [ ] oidc-middleware
- [x] okta-react
- [ ] okta-react-native
- [ ] okta-vue

## I'm submitting a:

- [x] Bug report  
- [ ] Feature request
- [ ] Other (Describe below)

## Current behavior

React 16.9.0 generates a warning for using `componentWillMount()` as this lifecycle event will be removed in future updates to React.

## Expected behavior

The RenderWrapper component should be updated according to React's migration guide at [React v16.9.0 and the Roadmap Update](https://reactjs.org/blog/2019/08/08/react-v16.9.0.html).

## Minimal reproduction of the problem with instructions

Upgrade react and react-dom on any existing project using React from 16.8.0 to 16.9.0.

## Environment

- Package Version: 1.2.0
- Browser: Chrome
- OS: MacOS 10.14.6
- Node version (`node -v`): 10.16.0
- Other:

**Comment by austinratcliff** [comment]
I want to +1 the bug above, as I ran into it today. I also noticed that my test that renders the component that uses `okta-react` returns this console error:

```
console.error node_modules/react-dom/cjs/react-dom.development.js:545
      Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
          in SecureRoute (created by Secure)
          in Secure (created by withAuth(SecureRoute))
          in withAuth(SecureRoute) (at App.js:17)
          in div (created by Security)
          in Security (created by Context.Consumer)
          in withRouter(Security) (at App.js:12)
          in Router (created by BrowserRouter)
          in BrowserRouter (at App.js:11)
          in div (at App.js:10)
          in app (at App.test.js:7)
```

I'm assuming that it's related, but I'm a React newbie, so I'm not 100% for sure.

**React and React DOM version**: 16.9.0
**Node version**: 12.7.0

**Comment by Fieel** [comment]
It's harmless but so annoying! I like my console windows clean without warning and errors ahah. Here's my case:

React and React DOM version: 16.9.0
Node version: v10.16.3

![image](https://user-images.githubusercontent.com/11167635/65691223-c48c5180-e070-11e9-9166-857197fe887f.png)

**Comment by dankolesnikov** [comment]
Same here, I prefer a clean console and we work diligently to avoid warning/errors. Okta engineers, can we see this fixed soon? I believe this should be a minor refactor changes.

**Comment by swiftone** [comment]
Just updating folks that we continue to see these messages and repeat our agreement to our managers (and they, in turn, agree as well, but there is always work) . Can't say when yet, but definitely in our conversations.

**Comment by swiftone** [comment]
for internal reference: OKTA-245540

**Comment by llamaforceone** [comment]
Upvoting as well. Would be great if this could be prioritized 🙏

**Comment by swiftone** [comment]
Just commenting that this remains on our radar.  I can't promise a release date yet, and I recommend people continue to upvote so it doesn't slide back, but I wanted people to know their voices are heard on this.

**Comment by miguelguadarrama** [comment]
I wish this was a _switftone_

Also +1 to keeping the console clean

**Comment by romerojunior** [comment]
Same behaviour running version 1.2.3, the SecureRoute component needs some work.

**Comment by edward93** [comment]
Would you guys accept a PR or you are already working on it?

**Comment by mraible** [comment]
@edward93 PRs are always encouraged!

**Comment by swiftone** [comment]
@edward93 - PRs are always welcome (see https://github.com/okta/okta-oidc-js/blob/master/CONTRIBUTING.md).  

React will get some dedicated love soon but we have not yet started on this particular task.

**Comment by edward93** [comment]
@swiftone thanks a lot.

**Comment by edward93** [comment]
Hi @swiftone could you please help me understand why those tests are failing for this PR #621 ? 
It seems kinda strange. 
Thanks in advance

**Comment by kerosan** [comment]
![image](https://user-images.githubusercontent.com/10149363/70758844-ebdbec00-1d4c-11ea-8ccf-afa9a97e2fe9.png)

**Comment by swiftone** [comment]
@edward93 - Thanks for the PR.  That build fails because the tests don't know who you are and thus your account can't access the CI tests.  Let me migrate your PR and get back to you.

**Comment by swiftone** [comment]
@edward93 - copied your changes into https://github.com/okta/okta-oidc-js/pull/635

**Comment by edward93** [comment]
@swiftone oh Thanks a lot. It seems like all checks are passed it only needs a reviewer. 
Can I add you as a reviewer?

**Comment by edward93** [comment]
@swiftone it seems like I can't update the PR since you are the owner now so I would highly appreciate if you could kindly add reviewers and check the checkbox of `No breaking changes` (during the migration it was unchecked somehow).

**Comment by swiftone** [comment]
@edward93 - Yes, leaving the "No braking changes" undefined was deliberate on my part.  We're currently inspecting the chance that some user is overriding .componentWillMount() and what would happen if they updated the library and got your change while doing so.  

library maintenance is fun! :) .

**Comment by edward93** [comment]
@swiftone OK got it. Thanks 👍

**Comment by swiftone** [comment]
Coming back to this post-holidays - someone is reviewing the PR for approval and merging.

**Comment by swiftone** [comment]
The PR to address this issue (by @edward93) has been merged and okta-react 1.3.1 has been released.
